src/lib/realmdb/RealmDB.js

```diff
@@ -38,30 +39,16 @@ class RealmDB implements DB, ProfileDB {
    * basic initialization
    * @param privateKey
    */
-  async init(privateKey: TextileCrypto.PrivateKey) {
+  async init(db: ThreadDB) {
     try {
-      this.db = new Database(
-        `feed_${privateKey.public.toString()}`,
-        {
-          name: 'Feed',
-          schema: FeedItemSchema,
-          indexes: [{ path: 'date' }, { path: 'data.hashedCode' }],
-        },
-        {
-          name: 'Assets',
-          schema: AssetsSchema,
-        },
-        {
-          name: 'Profiles',
-          schema: ProfilesSchema,
-        },
-      )
-      this.privateKey = privateKey
-      await this.db.open(3) // Versioned db on open
-      this.Feed = this.db.collection('Feed')
-      this.Feed.table.hook('updating', (modify, id, event) => this._notifyChange({ modify, id, event }))
-      this.Feed.table.hook('creating', (id, event) => this._notifyChange({ id, event }))
+      const { privateKey, Feed } = db
+
+      Feed.table.hook('creating', (id, event) => this._notifyChange({ id, event }))
+      Feed.table.hook('updating', (modify, id, event) => this._notifyChange({ modify, id, event }))
+
+      assign(this, { privateKey, db })
```

Comment:
the old syntax was this.Feed so you should assign Feed

Reply:
we agreed with this.db.Feed
i won't keep a thread db collection reference inside a realm db wrapper class

Reply:
ok